packages/details/test/details.test.js

```diff
@@ -96,6 +108,17 @@ describe('vaadin-details', () => {
         expect(details.opened).to.be.false;
       });
 
+      it(`should toggle opened on ${type} summary child click`, () => {
```

Comment:
Should every child be able toggle the details or should there be some kind of exception for interactive components like e.g. buttons that can be added to the summary? 

Example: 
A detail summary which contains text and a button with an question mark icon on the right side which can people click to get more information about the content of the details without opening it.

Reply:
I think it should at least be possible to make those exceptions manually, if we don't make them by default.